swarm/src/connection.rs

```diff
@@ -1000,7 +991,9 @@ mod tests {
             self: Pin<&mut Self>,
             _: &mut Context<'_>,
         ) -> Poll<Result<Self::Substream, Self::Error>> {
-            Poll::Ready(Ok(PendingSubstream(Arc::downgrade(&self.counter))))
+            Poll::Ready(Ok(PendingSubstream {
+                _weak: Arc::downgrade(&self.counter),
```

Comment:
clippy reported `PendingSubstream`s `Weak<()>` was not being read and suggested replacing to `PendingSubstream(())` but removing the weak reference fails tests, I used this pattern instead.